Userland/Libraries/LibELF/DynamicLinker.cpp

```diff
@@ -260,61 +285,7 @@ static int __dl_iterate_phdr(DlIteratePhdrCallbackFunction callback, void* data)
 
 static void initialize_libc(DynamicObject& libc)
 {
-    // Traditionally, `_start` of the main program initializes libc.
-    // However, since some libs use malloc() and getenv() in global constructors,
-    // we have to initialize libc just after it is loaded.
-    // Also, we can't just mark `__libc_init` with "__attribute__((constructor))"
-    // because it uses getenv() internally, so `environ` has to be initialized before we call `__libc_init`.
-    auto res = libc.lookup_symbol("environ"sv);
-    VERIFY(res.has_value());
-    *((char***)res.value().address.as_ptr()) = s_envp;
-
-    // __stack_chk_guard should be initialized before anything significant (read: global constructors) is running.
-    // This is not done in __libc_init, as we definitely have to return from that, and it might affect Loader as well.
-    res = libc.lookup_symbol("__stack_chk_guard"sv);
-    VERIFY(res.has_value());
-    void* stack_guard = res.value().address.as_ptr();
-    arc4random_buf(stack_guard, sizeof(uintptr_t));
-
-#ifdef AK_ARCH_64_BIT
-    // For 64-bit platforms we include an additional hardening: zero the first byte of the stack guard to avoid
-    // leaking or overwriting the stack guard with C-style string functions.
-    ((char*)stack_guard)[0] = 0;
-#endif
-
-    res = libc.lookup_symbol("__environ_is_malloced"sv);
-    VERIFY(res.has_value());
-    *((bool*)res.value().address.as_ptr()) = false;
-
-    res = libc.lookup_symbol("exit"sv);
-    VERIFY(res.has_value());
-    s_libc_exit = (LibCExitFunction)res.value().address.as_ptr();
```

Comment:
is s_libc_exit unused in loader code? 

similar q with __environ_is_malloced. Do we not care to set it to false here b/c it's default init to false?

Reply:
> is s_libc_exit unused in loader code?

Well, it still compiles and I wasn't able to find anything that uses it.

> Do we not care to set it to false here b/c it's default init to false?

Yes.

Reply:
At the same time I made __environ_is_malloced weak for some reason, let me fix that up